plugins/woocommerce/tests/api-core-tests/README.md

```diff
@@ -130,25 +133,309 @@ After you run a test, it's best to restart the environment to start from a fresh
+```js
+			expect( id ).toBeDefined();
+			expect( code ).toEqual( expectedCouponCode );
+```
+
+Verify [field types](https://github.com/woocommerce/woocommerce/blob/d19c20491e5a7ade64c8fd530f01e0f3f3f7e29c/plugins/woocommerce/tests/api-core-tests/tests/shipping/shipping-method.test.js#L85)
+```js
+expect(typeof responseJSON.id).toEqual('string');
+```
+
+Verify [field values](https://github.com/woocommerce/woocommerce/blob/4ac1d822ac9082102536b0f7aa9cb0553965adaa/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L186) 
+```js
+expect( updatedCoupons[ 1 ].amount ).toEqual( '25.00' );
+```
+
+Verify [field length](https://github.com/woocommerce/woocommerce/blob/4ac1d822ac9082102536b0f7aa9cb0553965adaa/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L136)
+```js
+expect( actualCoupons ).toHaveLength( expectedCoupons.length );
+```
+
+Verify search [headers](https://github.com/woocommerce/woocommerce/blob/d19c20491e5a7ade64c8fd530f01e0f3f3f7e29c/plugins/woocommerce/tests/api-core-tests/tests/orders/orders.test.js#L2703)
+```js
+// Verify total page count.
+			expect( page1.headers()[ 'x-wp-total' ] ).toEqual(
+				ORDERS_COUNT.toString()
+			);
+			expect( page1.headers()[ 'x-wp-totalpages' ] ).toEqual( '3' );
+  ```
+
+Verify variable [not undefined](https://github.com/woocommerce/woocommerce/blob/778cb130f27d0dd0dc7da1acb0e89762f81c0f18/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L31)
+```js
+expect( couponId ).toBeDefined();
+```
+
+Verify [response contains an object](https://github.com/woocommerce/woocommerce/blob/778cb130f27d0dd0dc7da1acb0e89762f81c0f18/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L73)
+```js
+expect( await response.json() ).toEqual(
+			expect.objectContaining( updatedCouponDetails )
+		);
+```
+
+Verify [response contains an array containing an object](https://github.com/woocommerce/woocommerce/blob/778cb130f27d0dd0dc7da1acb0e89762f81c0f18/plugins/woocommerce/tests/api-core-tests/tests/coupons/coupons.test.js#L388)
+```js
+		expect( responseJSON.coupon_lines[ 0 ].meta_data ).toEqual(
+			expect.arrayContaining( [
+				expect.objectContaining( {
+					key: 'coupon_data',
+					value: expect.objectContaining( {
+						code: testCoupon.code,
+					} ),
+				} ),
+			] )
+		);
+```
+
+## Debugging tests
+
+The Playwright debugger won't work for the API tests as it is based around GUI interactions.
+
+For now it is simple enough to add `console.log()` statements to output the values of your response/JSON/variables/status etc. Be sure to remove them when done ;) 
+
+You can also use the handy [REST API Log](https://wordpress.org/plugins/wp-rest-api-log/) plugin to see the API request information within WordPress. It displays the details, request headers, query params, body params, body content, response headers and response body information.
```

Comment:
Good call to suggest this plugin! Very helpful indeed when debugging.

Reply:
Glory goes to @zhongruige on this suggestion! I had a play about with the plugin and it looked great.